index.js

```diff
@@ -174,20 +173,26 @@ function parseStyles(
 ) {
   var statements = parseStatements(result, styles)
 
-  return Promise.resolve(statements).then(promiseEach(function(stmt) {
-    stmt.media = joinMedia(media, stmt.media || [])
+  return Promise.resolve(statements).then(function(stmts) {
+    return stmts.reduce(function(promise, stmt) {
+      return promise.then(function() {
```

Comment:
I’ll add a comment. I’m actually glad to see the `reduce` and `then` terminology used here instead of `each`, because `each` did not make me aware that this was processing the statements in sequence.